app/views/scenarios/_confirm_deletion_modal.html.erb

```diff
@@ -0,0 +1,46 @@
+<div id="confirm-scenario-deletion-<%= scenario.id %>" class="modal fade" tabindex="-1" role="dialog" aria-hidden="true">
+  <div class="modal-dialog">
+    <div class="modal-content">
+      <%= form_for(scenario, url: scenario_path(scenario), method: :delete) do |f| %>
+        <div class="modal-header">
+          <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
+          <h4 class="modal-title">
+            <% if scenario == @scenario %>
```

Comment:
I think this works because in once case `@scenario` is `nil`?  Might be clearer as `if @scenario && scenario == @scenario`